client/my-sites/domains/domain-search/site-redirect-step.jsx

```diff
@@ -99,43 +98,68 @@ var SiteRedirectStep = React.createClass( {
 
 			this.addSiteRedirectToCart( domain );
 		}.bind( this ) );
-	},
+	};
 
-	addSiteRedirectToCart: function( domain ) {
+	addSiteRedirectToCart = ( domain ) => {
 		upgradesActions.addItem( cartItems.siteRedirect( { domain: domain } ) );
+		page( '/checkout/' + this.props.selectedSite.slug );
+	};
 
-		if ( this.isMounted() ) {
-			page( '/checkout/' + this.props.selectedSite.slug );
-		}
-	},
-
-	getValidationErrorMessage: function( domain, error ) {
+	getValidationErrorMessage = ( domain, error ) => {
 		switch ( error.code ) {
 			case 'invalid_domain':
-				return this.translate( 'Sorry, %(domain)s does not appear to be a valid domain name.', {
+				return this.props.translate( 'Sorry, %(domain)s does not appear to be a valid domain name.', {
```

Comment:
A nice way to shorten this is to extract `translate` with destructuring on top of the function like:
```const { translate } = this.props;```
and then just use `translate`.